mocca-client/src/main/java/com/paypal/mocca/client/MoccaDeserializer.java

```diff
@@ -38,10 +38,10 @@
      * @return an optional with the deserialized and bound response object, which can be a list or not
      */
     Optional<?> deserialize(final InputStream inputStream, final Type responseType, final String operationName) {
-        return MoccaUtils.getInnerType(responseType, List.class)
-                .map(MoccaUtils::erase)
+        return MoccaReflection.getInnerType(responseType, List.class)
```

Comment:
we might want to add a task in the backlog to look at this for native vm.  By using reflection we will make this client library not directly usable on GraalVM

Reply:
I have just done a super quick research on this, and **it sounds** (I am still not sure) like GraalVM does support reflection, although it might get tricky. Anyways, I think we should definitely test Mocca in GraalVM and see if it works.

Please feel free to open the issue.

Reply:
Never mind. I have just opened it. #37